src/Google.Api.CommonProtos/TypeExtensions/Date.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+
+namespace Google.Type
+{
+    /// <summary>
+    /// Extensions to <see cref="Date"/>.
+    /// </summary>
+    public partial class Date
+    {
+        /// <summary>
+        /// Converts <see cref="Date"/> to <see cref="DateTime"/>.
+        /// </summary>
+        /// <returns>The converted <see cref="DateTime"/> with no time specified.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when <see cref="Date.Year"/>, <see cref="Date.Month"/>, and/or <see cref="Date.Day"/> are not within the valid range.</exception>
+        public DateTime ToDateTime()
+        {
+            try
+            {
+                return new DateTime(Year, Month, Day);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new InvalidOperationException($"Cannot convert to {nameof(DateTime)} because {nameof(Year)}, {nameof(Month)}, and/or {nameof(Day)} must be within the valid range for a {nameof(DateTime)}.");
+            }
+        }
+
+        /// <summary>
+        /// Converts <see cref="Date"/> to <see cref="DateTimeOffset"/>.
+        /// </summary>
+        /// <returns>The converted <see cref="DateTimeOffset"/> with no time specified and an Offset of <see cref="TimeSpan.Zero"/>.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when <see cref="Date.Year"/>, <see cref="Date.Month"/>, and/or <see cref="Date.Day"/> are not within the valid range.</exception>        
+        public DateTimeOffset ToDateTimeOffset()
+        {
+            try
+            {
+                return new DateTimeOffset(Year, Month, Day, 0, 0, 0, TimeSpan.Zero);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new InvalidOperationException($"Cannot convert to {nameof(DateTimeOffset)} because {nameof(Year)}, {nameof(Month)}, and/or {nameof(Day)} must be within the valid range for a {nameof(DateTimeOffset)}.");
+            }
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Date"/> instance from the <see cref="DateTime.Date"/> part of <see cref="DateTime"/>.
+        /// </summary>     
+        /// <param name="dateTimeToConvert">The <see cref="DateTime"/> instance being converted.</param>
+        /// <returns>The created <see cref="Date"/>.</returns>
+        public static Date CreateDate(DateTime dateTimeToConvert) =>
```

Comment:
Rename to FromDateTime to match the https://github.com/google/protobuf/blob/master/csharp/src/Google.Protobuf/WellKnownTypes/TimeExtensions.cs?
(Ditto method below)